flink-scala-shell/src/test/scala/org/apache/flink/api/scala/ScalaShellITCase.scala

```diff
@@ -322,32 +320,20 @@ object ScalaShellITCase {
 
   @BeforeClass
   def beforeAll(): Unit = {
-    val isNew = TestBaseUtils.isNewCodebase()
-    if (isNew) {
-      configuration.setString(CoreOptions.MODE, CoreOptions.NEW_MODE)
-      // set to different than default so not to interfere with ScalaShellLocalStartupITCase
-      configuration.setInteger(RestOptions.PORT, 8082)
-      val miniConfig = new MiniClusterConfiguration.Builder()
-        .setConfiguration(configuration)
-        .setNumSlotsPerTaskManager(parallelism)
-        .build()
-
-      val miniCluster = new MiniCluster(miniConfig)
-      miniCluster.start()
-      port = miniCluster.getRestAddress.getPort
-      hostname = miniCluster.getRestAddress.getHost
-
-      cluster = Some(Left(miniCluster))
-    } else {
-      configuration.setString(CoreOptions.MODE, CoreOptions.LEGACY_MODE)
-      configuration.setInteger(TaskManagerOptions.NUM_TASK_SLOTS, parallelism)
-      val standaloneCluster = new StandaloneMiniCluster(configuration)
-
-      hostname = standaloneCluster.getHostname
-      port = standaloneCluster.getPort
-
-      cluster = Some(Right(standaloneCluster))
-    }
+    configuration.setString(CoreOptions.MODE, CoreOptions.NEW_MODE)
+    // set to different than default so not to interfere with ScalaShellLocalStartupITCase
+    configuration.setInteger(RestOptions.PORT, 8082)
+    val miniConfig = new MiniClusterConfiguration.Builder()
+      .setConfiguration(configuration)
+      .setNumSlotsPerTaskManager(parallelism)
+      .build()
+
+    val miniCluster = new MiniCluster(miniConfig)
+    miniCluster.start()
+    port = miniCluster.getRestAddress.getPort
+    hostname = miniCluster.getRestAddress.getHost
+
+    cluster = Some(Left(miniCluster))
```

Comment:
do we still need the `Either` here?

Reply:
I don't think so. I will remove it and also remove the `StandaloneMiniCluster` at the same moment.